tests/functional/VestingEscrow/test_claim.py

```diff
@@ -1,54 +1,68 @@
-import pytest
-from brownie import ZERO_ADDRESS
+import brownie
 
 
-def test_claim_full(vesting, token, accounts, chain, end_time):
+def test_claim_full(deployed_vesting, token, recipient, chain, end_time, balance):
     chain.sleep(end_time - chain.time())
-    vesting.claim({"from": accounts[1]})
+    deployed_vesting.claim({"from": recipient})
 
-    assert token.balanceOf(accounts[1]) == 10 ** 20
+    assert token.balanceOf(recipient) == balance
 
 
-def test_claim_less(vesting, token, accounts, chain, end_time):
+def test_claim_less(deployed_vesting, token, recipient, chain, end_time, balance):
     chain.sleep(end_time - chain.time())
-    vesting.claim(accounts[1], vesting.total_locked() / 10, {"from": accounts[1]})
+    deployed_vesting.claim(
+        recipient, deployed_vesting.total_locked() / 10, {"from": recipient}
+    )
 
-    assert token.balanceOf(accounts[1]) == 10 ** 19
+    assert token.balanceOf(recipient) == balance / 10
 
 
-def test_claim_beneficiary(vesting, token, accounts, chain, end_time):
+def test_claim_beneficiary(
+    deployed_vesting, token, random_guy, recipient, chain, end_time, balance
+):
     chain.sleep(end_time - chain.time())
-    vesting.claim(accounts[2], {"from": accounts[1]})
+    deployed_vesting.claim(random_guy, {"from": recipient})
 
-    assert token.balanceOf(accounts[2]) == 10 ** 20
+    assert token.balanceOf(random_guy) == balance
 
 
-def test_claim_before_start(vesting, token, accounts, chain, start_time):
+def test_claim_before_start(deployed_vesting, token, recipient, chain, start_time):
     chain.sleep(start_time - chain.time() - 5)
-    vesting.claim({"from": accounts[1]})
+    deployed_vesting.claim({"from": recipient})
+
+    assert token.balanceOf(recipient) == 0
+
+
+def test_claim_after_end(deployed_vesting, token, recipient, chain, end_time, balance):
+    chain.sleep(end_time - chain.time() + 100)
+    deployed_vesting.claim({"from": recipient})
 
-    assert token.balanceOf(accounts[1]) == 0
+    assert token.balanceOf(recipient) == balance
 
 
-def test_claim_partial(vesting, token, accounts, chain, start_time, end_time):
-    chain.sleep(vesting.start_time() - chain.time() + 31337)
-    tx = vesting.claim({"from": accounts[1]})
+def test_claim_partial(deployed_vesting, token, recipient, chain, start_time, end_time):
+    chain.sleep(deployed_vesting.start_time() - chain.time() + 31337)
+    tx = deployed_vesting.claim({"from": recipient})
     expected_amount = (
-        vesting.total_locked() * (tx.timestamp - start_time) // (end_time - start_time)
+        deployed_vesting.total_locked()
```

Comment:
I suggest avoiding using calls to contract in such cases and using `balance` or a similar fixture to isolate and make a more predictive test case.

Reply:
Removed